src/analysis/convex.lean

```diff
@@ -412,18 +412,51 @@ begin
   simpa [hab, zero_lt_one] using h hx hy ha hb,
 end⟩
 
-lemma convex_on_linorder [linear_order α] {f : α → ℝ} : convex_on D f ↔
-  convex D ∧ ∀ {x y : α} {a b : ℝ}, x ∈ D → y ∈ D → x < y → 0 ≤ a → 0 ≤ b → a + b = 1 →
-    f (a • x + b • y) ≤ a * f x + b * f y :=
+/-- For functions on linear ordered sets, it suffices to verify the requirements for `x < y`
```

Comment:
This docstring can not be understood just by itself. You say "it suffices", but you don't say what you want to do. Add "to prove convexity" just before "it suffices", for instance, to make it self-containted.

Reply:
Tried to fix.